src/AppKit/NSOpenGLPixelFormat.cs

```diff
@@ -54,67 +54,94 @@ public NSOpenGLPixelFormat (NSOpenGLPixelFormatAttribute[] attribs) : base (NSOb
 			}
 		}
 
-		static NSOpenGLPixelFormatAttribute [] ConvertToAttributes (object [] args)
+		public NSOpenGLPixelFormat (uint [] attribs) : base (NSObjectFlag.Empty)
 		{
-			var list = new List<NSOpenGLPixelFormatAttribute> ();
-			for (int i = 0; i < args.Length; i++){
-				var v = (NSOpenGLPixelFormatAttribute) args [i];
-				switch (v){
-				case NSOpenGLPixelFormatAttribute.AllRenderers:
-				case NSOpenGLPixelFormatAttribute.DoubleBuffer:
-				case NSOpenGLPixelFormatAttribute.Stereo:
-				case NSOpenGLPixelFormatAttribute.MinimumPolicy:
-				case NSOpenGLPixelFormatAttribute.MaximumPolicy:
-				case NSOpenGLPixelFormatAttribute.OffScreen:
-				case NSOpenGLPixelFormatAttribute.FullScreen:
-				case NSOpenGLPixelFormatAttribute.SingleRenderer:
-				case NSOpenGLPixelFormatAttribute.NoRecovery:
-				case NSOpenGLPixelFormatAttribute.Accelerated:
-				case NSOpenGLPixelFormatAttribute.ClosestPolicy:
-				case NSOpenGLPixelFormatAttribute.Robust:
-				case NSOpenGLPixelFormatAttribute.BackingStore:
-				case NSOpenGLPixelFormatAttribute.Window:
-				case NSOpenGLPixelFormatAttribute.MultiScreen:
-				case NSOpenGLPixelFormatAttribute.Compliant:
-				case NSOpenGLPixelFormatAttribute.PixelBuffer:
+			if (attribs == null)
+				throw new ArgumentNullException ("attribs");
+
+			unsafe
+			{
+				uint [] copy = new uint [attribs.Length + 1];
+				Array.Copy (attribs, 0, copy, 0, attribs.Length);
+
+				fixed (uint* pArray = copy) {
+					if (IsDirectBinding) {
+						Handle = XamCore.ObjCRuntime.Messaging.IntPtr_objc_msgSend_IntPtr (this.Handle, selInitWithAttributes, new IntPtr ((void*)pArray));
```

Comment:
Call `InitializeHandle (...)` instead of the `Handle` setter (it will throw a much better exception if the native init method returns NULL).